### ISSUE TITLE
Facebook login succeeds but the plugin throws an error at iOS

### ISSUE THREAD

**Issue by kaansoral** [Wizcorp/phonegap-facebook-plugin#479]
I've inspected the app with XCODE - no error is logged

I know the auth succeeds because facebook pings the url with the fb{appid}://authorize?access_token (by the way I removed the custom url scheme plugin too, it wasn't the source of the issue)

Long story short, the auth succeeds however the plugin claims "An error occurred"

**Comment by phuongnm3** [comment]
Hi @andrewb273   have you resolved this issue? we also run across this error but cannot resolve so far.

Really need your help :(

**Comment by andrewb273** [comment]
Hi @phuongnm3, I haven't been able to resolve the issue yet.

Recent bug testing suggest potential conflicting information between FB.getAuthResponse() and FB.api('/me') in terms of the user ID, so when I get a chance I'll be investigating that further.

**Comment by phuongnm3** [comment]
Thanks @andrewb273  I'm not sure if many people run across this issue but I was stuck with this one for a week. Do you have any suggestion? I really need to resolve it, my application will need to push out production next week :(

**Comment by andrewb273** [comment]
@phuongnm3 We hide front page FB login from our iOS production builds, so no decent suggestions! Though do look for that anomaly that I described.

Also test to get a better idea of the frequency of this issue... i.e. different accounts, fresh accounts etc. We find this issue isn't too common, we actually enable "link your account to FB" in the user's settings menu.

If you get some time, do some more bug testing to track it through the plugin code. The more people that do detailed debugging, the more likely we can get a fix to this issue soon.

**Comment by phuongnm3** [comment]
@andrewb273  I have same issue with you, same location and error. The feature works only with the fb app owner account, the other account will get into this error.

Unfortunately, it is very common with me. I cannot find "link your account to FB" , could you give me a screen.

**Comment by andrewb273** [comment]
@phuongnm3 Ok that's good I suppose. Though if you're keen, track it further - FacebookSDK for iOS is a [repo on GitHub](https://github.com/facebook/facebook-ios-sdk).

Also can you please check the results of .getAuthResponse() and .api('/me') in your JS, and compare the values?

"Link your account to FB" is just a common app feature that we've manually implemented - simply means tying a user's account to a Facebook account.

**Comment by phuongnm3** [comment]
@andrewb273  I had error with .login() so I cannot call  .getAuthResponse() and .api('/me')  

I will try replacing with latest version of SDK and change to another app id to see. Hopefully phonegap-facebook-plugin team will look into it. I'm willing to provide any information/credential/source code that they can help me resolve the error :( This one is very critical with me.

Can I have your contact like email or something, I would like to see your configuration on fb app.

Thanks,
Phuong

**Comment by phuongnm3** [comment]
Hi @andrewb273 I just found a problem, I think it's root cause. The error occurs because previous fb user has not been cleaned. I see that previous fb id still exists in the error message, if I use previous fb user I can login. Do you know how to clean up fb user from app when they signs out our ios app?

**Comment by andrewb273** [comment]
@phuongnm3 What timezone are you? I'm GMT+11. It's 10 PM here so won't be working much longer. Check my profile, I've temporarily added an email to contact me via. Email me once you've got it.

Yeah clearing previous FB user - I've frequently suspected that something like that could be causing the issue. Unfortunately FB doesn't provide a simple logout feature. FB.logout() logs the user not only out of your app, but out of FB itself. At a guess, this is something the plugin should handle.

If you have ample time to devote to this, I suggest building a vanilla ObjC app and incorporating FacebookSDK as part of your testing. Would give you a standard on how it should work. Though you'll need to have some ObjC experience for that to be a worthwhile endeavour...

**Comment by andrewb273** [comment]
Hey @phuongnm3, when you can, describe the solution here that you mentioned to me which seems to be currently resolving this problem.

**Comment by phuongnm3** [comment]
yeah, solution is that you need to call facebookConnectPlugin.logout(Function success, Function failure) before you call facebookConnectPlugin.login(), this is to clean up last user in the cache somehow.

**Comment by poyannabati** [comment]
Thanks @phuongnm3, calling facebookConnectPlugin.logout() works nicely.

**Comment by felquis** [comment]
First I figured out about this problem manually doing the change from #958 then I use logout before login as @kaansoral said. It works fine now, but I have no ideia of the implications of using logout before login.. :/ 

Thank you @kaansoral

**Comment by jorgevidal** [comment]
Thank you @phuongnm3 calling facebookConnectPlugin.logout() is the solution for this problem.
